partition/exceptions.py

```diff
@@ -12,3 +12,11 @@ class DuplicateFileError(Exception):
     def __init__(self, message):
         self.message = message
         super().__init__(self.message)
+
+
+class ClobberingFileError(Exception):
+    """ Custom exception class raised when encoutering files that would be overritten """
+
+    def __init__(self, message):
```

Comment:
To store the list of files as a list, but still allow this to behave like standard exceptions, change to this:

```python
def __init__(self, *args, files=None, **kwargs):
    super().__init__(*args, **kwargs)
    self.files = files or []
```